default.nix

```diff
@@ -0,0 +1,21 @@
+let
+  flakeLock = builtins.fromJSON (builtins.readFile ./flake.lock);
+  fetchFromGitHub = input:
+    with flakeLock.nodes.${input}.locked;
+    assert type == "github";
+      fetchTarball {
+        url = "https://github.com/${owner}/${repo}/archive/${rev}.tar.gz";
+        sha256 = narHash;
```

Comment:
This re-implements [the lock-file parser in `flake-compat`](https://github.com/nix-community/flake-compat/blob/8bf105319d44f6b9f0d764efa4fdef9f1cc9ba1c/default.nix#L16-L77).

Unfortunately that code is not exported. Maybe fix upstream first by exporting that lock-file parser, and then [get `flake-compat` idiomatically](https://github.com/nix-community/flake-compat/blob/8bf105319d44f6b9f0d764efa4fdef9f1cc9ba1c/README.md#L58-L71).

Reply:
There's apparently [a copy of that code in dream2nix](https://github.com/nix-community/dream2nix/blob/07bd44a3365efdbf4d17a14190636667e5fd3d48/dev-flake/flake-compat.nix#L21-L114)